src/transformStateWithClones.js

```
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const resultAllActions = [];
+  let stateCopy = { ...state };
+
+  for (const action of actions) {
+    const { type, extraData, keysToRemove } = action;
+
+    switch (type) {
+      case 'addProperties':
+        stateCopy = { ...stateCopy, ...extraData };
+        break;
+      case 'removeProperties':
+        for (const element of keysToRemove) {
+          delete stateCopy[element];
+        }
+        break;
+      case 'clear':
+        for (const key in stateCopy) {
+          delete stateCopy[key];
+        }
+        break;
+      default:
+        break;
```

This violates checklist item #3: 'switch/case should always have default case for error handling'. An empty `default` case causes unknown action types to be silently ignored. It's better practice to throw an error here to signal that an invalid action was passed to the function.